generics/helpers/solutionAndProjectTemplateUtils.js

```
@@ -138,16 +138,15 @@ function createSolution(solutionData, checkDate = false, userDetails, isExternal
 
-			let newprogramData
-			if (UTILS.convertStringToBoolean(isExternalProgram)) {
-				newprogramData = await surveyService.programsDocument(programMatchQuery, ['components'])
-			} else {
-				newprogramData = await programQueries.programsDocument(programMatchQuery, ['components'])
-			}
-			let updateProgram = await programQueries.findAndUpdate(programMatchQuery, {
-				$addToSet: {
-					components: { _id: solutionCreation._id, order: newprogramData[0].components.length + 1 },
-				},
+			UTILS.executeWithProgramLock(programData[0]._id.toString(), async () => {
```

ESLint

[error] 'UTILS' is not defined.

(no-undef)